pkg/agent/utils.go

```diff
@@ -497,6 +497,12 @@ func setCustomKubeletConfig(customKc *datamodel.CustomKubeletConfig,
 		if customKc.PodMaxPids != nil {
 			kubeletConfig.PodPidsLimit = to.Int64Ptr(int64(*customKc.PodMaxPids))
 		}
+		if customKc.ShutdownGracePeriod != nil {
```

Comment:
I also wanted to add some defaulting logic based on Kubernetes version. What do you think if we add `IsKubernetesVersionGe(xxx)` in this method? @UtheMan